packages/skygear-core/src/container.ts

```diff
@@ -131,6 +132,37 @@ export class AuthContainer<T extends BaseAPIClient> {
     this.parent.apiClient.accessToken = null;
   }
 
+  /**
+   * @internal
+   */
+  async _refreshAccessToken(): Promise<boolean> {
+    await this.parent.storage.delAccessToken(this.parent.name);
```

Comment:
Why do we need to delete it first?

Reply:
The access token is no longer valid at this point, so just delete it from storage.

Reply:
In case the refresh operation failed due to network error, the access token was deleted and the same flow cannot be retried because the client SDK will return error early complaining that access token is null.

Reply:
> the client SDK will return error early complaining that access token is null.

Could you point me to where the client SDK checks the access token is null? The client SDK should not do any check on access token at client-side, since it may stored in cookies.

In your case, the network error would be returned to consumer. If consumer retry the operation, the SDK would not send the access token to server. After receiving a 401 error, the SDK would attempt to refresh the token again.